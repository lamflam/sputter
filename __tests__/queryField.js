const snapshot = require("../testUtil/snapshot");
const fixtures = require("../testUtil/fixtures");

describe("Field query", () => {
  describe("single field", () => {
    snapshot(fixtures.conditional, ".test");
  });

  describe("field sequence", () => {
    snapshot(fixtures.simpleProgram, ".declarations.init");
  });
});
