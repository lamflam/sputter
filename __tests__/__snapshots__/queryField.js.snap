// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field query field sequence .declarations.init 1`] = `
> 1 | var x = 1;
    |         ^ NumericLiteral
  2 | var y = "y";
  3 | x = x * 2;
  4 | if (y) {
  1 | var x = 1;
> 2 | var y = "y";
    |         ^^^ StringLiteral
  3 | x = x * 2;
  4 | if (y) {
  5 |   y += "z";
`;

exports[`Field query single field .test 1`] = `
> 1 | if (x === 1) {
    |     ^^^^^^^ BinaryExpression
  2 |   foo();
  3 | } else {
  4 |   x = 2;
  4 |   x = 2;
  5 | }
> 6 | if ((x == "test" && true) || x) {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ LogicalExpression
  7 |   y = -1;
  8 | } else if (false) {
  9 |   y = 1;
   6 | if ((x == "test" && true) || x) {
   7 |   y = -1;
>  8 | } else if (false) {
     |            ^^^^^ BooleanLiteral
   9 |   y = 1;
  10 | }
  11 | 
`;
